concepts/observability-configurable-logging/01-groundwork.md

```diff
@@ -0,0 +1,47 @@
+
+# Requirements
+
+The basic requirements can be condensed to this list:
+
+- Basic backend configuration
+  - Have a vendor-neutral agent layer being responsible for collecting and shipping the telemetry data, but not permanently storing it (as a backend)
+  - Support configuration of the selected agent at runtime (no need to run a kyma upgrade process) in a scenario focused approach
+  - Support multiple configurations at the same time in individual resources to enable easy activation of dedicated scenarios
+  - one vendor-neutral input and output support is enough as a minimum, it should be possible to chain your custom agent for specific conversions (like for traces, supporting the OLTP protocol will support most of the vendors already. Chaining a custom otel-collector can do custom conversion to a specific protocol)
+  - all typical settings for supported outputs of the used agent are supported (do not hide/abstract them)
+  - the agent should run stable at anytime, bad configuration should be pre-validated and rejected, fast feedback is welcome
+  - secrets must be kept secret
+
+- Template definitions
+  - Have a mechanism to provide templates/best practices for typical scenarios which can be innstantiated at runtime
+  - Such template provides same feature-richness as a configuration scenario
+  - If specific templates are bundled with kyma, updates of a template in use should propogate to the actual agent configuration
+  - Supports placeholder definitions with default values and descriptions
+
+- Template instantiation
+  - A template gets instantiated by usually binding it to a secret, fullfilling placeholders like URL and credentials
+  - Placeholders can be satisfied with configMaps and static values as well
+  - It validates fullfilment of placeholders
+  - A template can be instantiated for specific workload and/or namespaces only
+
+# Proposed Solution
+
+The idea of the proposal is to introduce a pre-configured agent layer being responsible for collecting all telemetry data. That agents can be dynamically configured at runtime with different configuration scenarios so that the agents will start shipping the data to the configured backends. The dynamic configuration and management of the agent will happen by a new operator which can be configured via k8s resources. The agents and the new operator will be bundled in a new core package called `telemetry`. The bundled backends and UIs will become just one possible solution which can be installed optional and will stay in the existing modules like `logging`.
+
+![b](./assets/future_all.drawio.svg)
+
+# Focus on Logging
+
+While the section above was more general, in the following it will be focused on the logging aspect only. This is mainly to have an emerging approach and solving the most urgent topics first. As logs have the longest tradition, everyone is used and expects to have solutions for logging place.
```

Comment:
I know I reviewed that paragraph. Weird. Thanks for spotting this

```suggestion
While the section above was more general, in the following, it will be focused on the logging aspect only. This is mainly to have an emerging approach and solving the most urgent topics first. As logs have the longest tradition, everyone is used and expects to have solutions for logging in place.
```